lib/CL/pocl_cl.h

```diff
@@ -164,16 +164,27 @@ typedef struct pocl_kernel_metadata_s pocl_kernel_metadata_t;
 
 //############################################################################
 
+#define POCL_MAGIC_1 0xBE8906A1A83D8D23ULL
```

Comment:
What are these? Should be well-documented.

Reply:
This is still undocumented. Reading the code it's supposed to add extra 'validness' protection by checking if the magic numbers are there or not. Do you find this useful in some case/debugging? It is not nearly fool proof anyhow, like we've discussed in the past. If we allocate a new CL object to that space in another thread, the old reference looks valid even though it was freed in between etc. If you have found it useful, please add comments to the magic numbers or even rename to them to 'sentinel' or similar slightly more descriptive which is a common name for this type trickery.